projects/com.oracle.truffle.llvm.parser.factories/src/com/oracle/truffle/llvm/parser/factories/LLVMAggregateFactory.java

```diff
@@ -113,29 +114,29 @@ public static LLVMExpressionNode createInsertValue(LLVMAddressNode resultAggrega
         }
     }
 
-    public static LLVMExpressionNode createStructConstantNode(LLVMParserRuntime runtime, ResolvedType structType, boolean packed, ResolvedType[] types, LLVMExpressionNode[] constants) {
+    public static LLVMExpressionNode createStructConstantNode(LLVMParserRuntime runtime, Type structType, boolean packed, Type[] types, LLVMExpressionNode[] constants) {
         int[] offsets = new int[types.length];
         LLVMStructWriteNode[] nodes = new LLVMStructWriteNode[types.length];
         int currentOffset = 0;
-        int structSize = runtime.getTypeHelper().getByteSize(structType);
-        int structAlignment = runtime.getTypeHelper().getAlignmentByte(structType);
-        LLVMExpressionNode alloc = runtime.allocateFunctionLifetime(structType, structSize, structAlignment);
+        int structSize = ((LLVMTypeHelperImpl) runtime.getTypeHelper()).getByteSize(structType);
+        int structAlignment = ((LLVMTypeHelperImpl) runtime.getTypeHelper()).getAlignmentByte(structType);
+        LLVMExpressionNode alloc = runtime.allocateFunctionLifetime(LLVMToBitcodeAdapter.unresolveType(structType), structSize, structAlignment);
         for (int i = 0; i < types.length; i++) {
-            ResolvedType resolvedType = types[i];
+            Type resolvedType = types[i];
             if (!packed) {
-                currentOffset += runtime.getTypeHelper().computePaddingByte(currentOffset, resolvedType);
+                currentOffset += ((LLVMTypeHelperImpl) runtime.getTypeHelper()).computePaddingByte(currentOffset, resolvedType);
```

Comment:
Done.

This was a temporary solution to solve dependency problems across packages (https://github.com/graalvm/sulong/pull/489/commits/29a45d753fe4b6fa4208212ec55eafb5ff717e6d)